x/operator/keeper/genesis.go

```diff
@@ -0,0 +1,47 @@
+package keeper
+
+import (
+	"github.com/ExocoreNetwork/exocore/x/operator/types"
+	abci "github.com/cometbft/cometbft/abci/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+func (k Keeper) InitGenesis(ctx sdk.Context, state types.GenesisState) []abci.ValidatorUpdate {
+	for _, op := range state.Operators {
+		op := op
+		if err := k.SetOperatorInfo(ctx, op.EarningsAddr, &op); err != nil {
+			panic(err)
+		}
+	}
+	for _, record := range state.OperatorRecords {
+		addr := record.OperatorAddress
+		// #nosec G703 // already validated
+		operatorAddr, _ := sdk.AccAddressFromBech32(addr)
+		bootstrapping := false
+		for _, detail := range record.Chains {
+			// opt into the specified chain (TODO: avs address format)
+			if err := k.OptIn(ctx, operatorAddr, detail.ChainID); err != nil {
+				panic(err)
+			}
+			// #nosec G703 // already validated
+			key, _ := types.HexStringToPubKey(detail.ConsensusKey)
+			// then set pub key
+			if err := k.setOperatorConsKeyForChainID(
+				ctx, operatorAddr, detail.ChainID, key, true,
+			); err != nil {
+				panic(err)
+			}
+			bootstrapping = bootstrapping || ctx.ChainID() == detail.ChainID
+		}
+		if !bootstrapping {
+			// TODO: consider removing this check
```

Comment:
Should we consider multiple app chains here? It appears that this could lead to panics, especially if the inputs involve multiple app chains. Would it be simpler and more effective to only support the Exocore chain here? Then the other app chain AVSs could be launched after Exocore has been launched

Reply:
I see your point. I don't feel too strongly to keep this `panic` at this point, however, I added it to keep it future proof.

The `InitGenesis` function is designed (by the Cosmos-SDK) to support actual genesis as well as chain restarts and upgrades (that are not happening via governance). In the last two cases, it may be possible that app chains are imported to the state from the JSON file.

What do you think?

Reply:
Actually, this panic should not be here at all. It is not the responsibility of the operator module to validate this. It is something done by the dogfood module. I will move that check there.